cluster/addons/cluster-monitoring/influxdb/heapster-controller.json

```diff
@@ -0,0 +1,54 @@
+{
+  "apiVersion": "v1beta3",
+  "kind": "ReplicationController",
+  "metadata": {
+    "labels": {
+      "name": "heapster",
+      "kubernetes.io/cluster-service": "true"
+    },
+    "name": "monitoring-heapster-controller"
+  },
+  "spec": {
+    "replicas": 1,
+    "selector": {
+      "name": "heapster"
+    },
+    "template": {
+      "metadata": {
+        "labels": {
+          "name": "heapster",
+          "kubernetes.io/cluster-service": "true"
+        }
+      },
+      "spec": {
+        "containers": [
+          {
+            "image": "gcr.io/google_containers/heapster:v0.11.0",
+            "name": "heapster",
+            "env": [
+              {
+                "name": "FLAGS",
+                "value": "--poll_duration=2m --stats_resolution=1m --sink influxdb:http://$MONITORING_INFLUXDB_SERVICE_HOST:$MONITORING_INFLUXDB_PORT_8086_TCP_PORT"
```

Comment:
The heapster pod is supposed to handle backend failures. So if the dns name
does not resolve, it will retry internally before deciding to crash/fail.
Crashing without retires might confuse users.

On Tue, May 5, 2015 at 2:30 PM, Saad Ali <notifications@github.com> wrote:

> In cluster/addons/cluster-monitoring/influxdb/heapster-controller.json
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/7751#discussion_r29715109>
> :
>
> > +    "template": {
> > +      "metadata": {
> > +        "labels": {
> > +          "name": "heapster",
> > +          "kubernetes.io/cluster-service": "true"
> > +        }
> > +      },
> > +      "spec": {
> > +        "containers": [
> > +          {
> > +            "image": "gcr.io/google_containers/heapster:v0.11.0",
> > +            "name": "heapster",
> > +            "env": [
> > +              {
> > +                "name": "FLAGS",
> > +                "value": "--poll_duration=2m --stats_resolution=1m --sink influxdb:http://$MONITORING_INFLUXDB_SERVICE_HOST:$MONITORING_INFLUXDB_PORT_8086_TCP_PORT"
>
> If that is the case, there would be no guarantee that Kube-DNS would
> resolve monitoring-influxdb correctly--i.e. if heapster pod gets created
> before the heapster service, Kube-DNS would not have a entry for
> monitoring-influxdb--we would end up with the same problem.
>
> With the env variables approach, if the pod comes up before the service,
> and the variables are not set, heapster will terminate with error--and
> Kubernetes will create a new container which should have the variables set.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/7751/files#r29715109>
> .
>